### PR TITLE
Add improvements to Tracing Test Case

### PR DESCRIPTION
**Pull request by nadundesilva**

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 1]
```diff
@@ -410,6 +431,16 @@ private void rewriteObservableFunctionBody(BIRFunction func, BIRPackage pkg, boo
             } else if (currentBB.terminator.kind == InstructionKind.CALL
                     || (currentBB.terminator.kind == InstructionKind.FP_CALL
                     && !((FPCall) currentBB.terminator).isAsync)) {
+                /*
+                 * Traps for errors needs to be injected for each call and fp call separately to avoid messing up the
+                 * line numbers in the stack trace shown when a panic is thrown.
+                 *
+                 * These panic traps are different from the traps added in rewriteObservableFunctionInvocations method,
+                 * in the sense that these report the error to the Observation covering the current function this body
+                 * belongs to. Also these does not cover the observable calls and fp calls (they are handled using the
```
Comment (tharinduwijewardane) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:440]:
```suggestion
                 * belongs to. Also these do not cover the observable calls and fp calls (they are handled using the
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 2]
```diff
@@ -554,6 +582,10 @@ private void rewriteObservableFunctionInvocations(BIRFunction func, BIRPackage p
                     fixErrorTable(func, currentBB, observeEndBB);
                 }
                 {
+                    /*
+                     * Adding panic traps for the invocations. These reports the error to the Observation covering
```
Comment (tharinduwijewardane) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:586]:
```suggestion
                     * Adding panic traps for the invocations. These report the error to the Observation covering
```

Comment (nadundesilva) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java]: Fixed in 8ee1b9f

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java (hunk 1 above)
Comment (nadundesilva) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java]: Fixed in 8ee1b9f

**Pull request merged**